src/back-end/lib/routers/auth.ts

```diff
@@ -287,24 +287,26 @@ async function establishSessionWithClaims(connection: Connection, request: Reque
   }
 
   let username = getString(claims, 'preferred_username');
+  const idpId = getString(claims, 'idp_id');
 
   // Strip the vendor/gov suffix if present.  We want to match and store the username without suffix.
   if ((username.endsWith('@' + VENDOR_IDP_SUFFIX) && userType === UserType.Vendor) || (username.endsWith('@' + GOV_IDP_SUFFIX) && userType === UserType.Government)) {
     username = username.slice(0, username.lastIndexOf('@'));
   }
 
-  if (!username || !tokenSet.access_token || !tokenSet.refresh_token) {
+  if (!username || !idpId || !tokenSet.access_token || !tokenSet.refresh_token) {
     throw new Error('authentication failure - invalid claims');
   }
 
-  const dbResult = await findOneUserByTypeAndUsername(connection, userType, username);
+  const dbResult = await findOneUserByIdpId(connection, idpId);
```

Comment:
Need to query by `idpId` and `userType` because `idpId`s aren't guaranteed to be unique across all users

Reply:
Right, done. Ty.